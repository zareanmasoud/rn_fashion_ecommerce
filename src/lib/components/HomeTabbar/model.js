// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
