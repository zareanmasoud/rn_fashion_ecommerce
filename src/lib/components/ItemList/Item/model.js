// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
