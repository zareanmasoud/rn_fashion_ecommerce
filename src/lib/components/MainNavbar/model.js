// @flow

export type Button = {
  action: Function,
  src: string,
};
