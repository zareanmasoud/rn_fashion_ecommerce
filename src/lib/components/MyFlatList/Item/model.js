// @flow

export type Data = {
  id: string,
  photo: string,
};
