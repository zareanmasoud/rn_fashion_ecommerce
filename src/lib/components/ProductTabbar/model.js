// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
