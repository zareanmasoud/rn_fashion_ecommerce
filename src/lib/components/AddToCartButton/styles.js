import {StyleSheet} from 'react-native';
import R from 'res/R';

const styles = StyleSheet.create({
  buttonView: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 5,
    paddingHorizontal: 20,
  },
  buttonText: {
    color: R.colors.text,
  },
});

export default styles;
