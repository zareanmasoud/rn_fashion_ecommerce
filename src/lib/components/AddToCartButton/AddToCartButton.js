import React from 'react';
import {Image, TouchableOpacity, View} from 'react-native';
import styles from './styles';

type Props = {
  handlePress: Function,
  src: string,
  // text: string,
};

export default function AddToCartButton({handlePress, src}: Props) {
  return (
    <TouchableOpacity testID="add_to_cart_button" onPress={() => handlePress()}>
      <View style={styles.buttonView}>
        <Image source={src} style={styles.buttonImg} />
      </View>
    </TouchableOpacity>
  );
}
