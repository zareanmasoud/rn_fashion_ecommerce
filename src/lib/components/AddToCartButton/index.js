import AddToCartButton from './AddToCartButton';

export {AddToCartButton as default};
