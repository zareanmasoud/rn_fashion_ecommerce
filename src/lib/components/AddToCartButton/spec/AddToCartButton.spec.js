import React from 'react';
import {shallow} from 'enzyme';
import AddToCartButton from '../AddToCartButton';
import model from './model';

const mockHandlePress = jest.fn();
mockHandlePress.mockReturnValue('Link on press invoked');
const enhancedModel = model.addToCart(mockHandlePress);

describe('AddToCartButton Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<AddToCartButton {...enhancedModel} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interaction', () => {
    describe('handlePress', () => {
      let wrapper;
      beforeEach(() => {
        wrapper = shallow(<AddToCartButton {...enhancedModel} />);
        jest.clearAllMocks();
      });

      it('should call handlePress', () => {
        wrapper
          .find({testID: 'add_to_cart_button'})
          .first()
          .props()
          .onPress();
        expect(mockHandlePress.mock.calls.length).toBe(1);
        expect(mockHandlePress).toHaveBeenCalled();
        expect(mockHandlePress).toHaveBeenCalledTimes(1);
      });

      // it('should call openURL if url is provided', () => {
      //   wrapper.handlePress();
      //   // 3. mockOpenURL should be called since we have passed in 'url' prop
      //   expect(mockOpenURL).toHaveBeenCalled();
      //   expect(mockOpenURL).toHaveBeenCalledTimes(1);
      //   expect(mockOpenURL).toHaveBeenCalledWith('https://www.test.com');
      // });
      //
      // it('should not call openURL if url is nor provided', () => {
      //   const innerInstance = shallow(
      //     <AddToCartButton label="test label" handlePress={mockHandlePress} />,
      //   ).instance();
      //   innerInstance.handlePress();
      //   // 4. mockOpenURL should NOT be called as we have NOT passed in 'url' prop
      //   expect(mockOpenURL).not.toHaveBeenCalled();
      // });
    });
  });
});
