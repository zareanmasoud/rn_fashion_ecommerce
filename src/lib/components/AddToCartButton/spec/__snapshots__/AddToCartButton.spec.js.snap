// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton Component Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  testID="add_to_cart_button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 42.857142857142854,
          "tintColor": "#4d4d4dff",
          "width": 42.857142857142854,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
